scrypto-test/src/ledger_simulator/compile.rs

```diff
@@ -2,37 +2,67 @@ use crate::prelude::*;
 use scrypto_compiler::*;
 use std::path::Path;
 
+pub enum CompileProfile {
+    /// Uses default compilation options - same as `scrypto build`. Should be used in all cases which requires
+    /// compilation results to be as close to production as possible (for instance costing related tests).
+    Standard,
+    /// Same as Standard with enabled all logs from error to trace level.
+    StandardWithTraceLogs,
+    /// Disables WASM optimization to speed-up compilation process, by default used by SDK PackageFactory.
+    Fast,
+    /// Disables WASM optimization and enables all logs from error to trace level, by default used by Ledger Simulator.
+    FastWithTraceLogs,
+}
+
 pub struct Compile;
 
 impl Compile {
-    pub fn compile<P: AsRef<Path>>(package_dir: P) -> (Vec<u8>, PackageDefinition) {
+    pub fn compile<P: AsRef<Path>>(
+        package_dir: P,
+        compile_profile: CompileProfile,
+    ) -> (Vec<u8>, PackageDefinition) {
         Self::compile_with_env_vars(
             package_dir,
             btreemap! {
                 "RUSTFLAGS".to_owned() => "".to_owned(),
                 "CARGO_ENCODED_RUSTFLAGS".to_owned() => "".to_owned(),
             },
+            compile_profile,
+            true,
         )
     }
 
     // required for compile-blueprints-at-build-time feature in radix-engine-tests
     pub fn compile_with_env_vars<P: AsRef<Path>>(
         package_dir: P,
         env_vars: sbor::rust::collections::BTreeMap<String, String>,
+        compile_profile: CompileProfile,
+        _use_coverage: bool,
     ) -> (Vec<u8>, PackageDefinition) {
         // Initialize compiler
         let mut compiler_builder = ScryptoCompiler::builder();
-        compiler_builder
-            .manifest_path(package_dir.as_ref())
-            .optimize_with_wasm_opt(None)
-            .log_level(Level::Trace); // all logs from error to trace
+        compiler_builder.manifest_path(package_dir.as_ref());
+
+        match compile_profile {
+            CompileProfile::Standard => (),
+            CompileProfile::StandardWithTraceLogs => {
+                compiler_builder.log_level(Level::Trace); // all logs from error to trace
+            }
+            CompileProfile::Fast => {
+                compiler_builder.optimize_with_wasm_opt(None);
+            }
+            CompileProfile::FastWithTraceLogs => {
+                compiler_builder.optimize_with_wasm_opt(None);
+                compiler_builder.log_level(Level::Trace); // all logs from error to trace
+            }
+        }
 
         env_vars.iter().for_each(|(name, value)| {
             compiler_builder.env(name, EnvironmentVariableAction::Set(value.clone()));
         });
 
         #[cfg(feature = "coverage")]
-        {
+        if _use_coverage {
```

Comment:
Do we still need to feature flag?

Reply:
Yes, because it changes some limits to larger values in transaction executor and wasm executor. There was missing `coverage` feature in `scrypto-test` project.
I also fixed `scrypto coverage` memory error which was caused by using coverage feature and Rustflags on 1st compilation stage (schema extract).